### PR TITLE
More systemd-oomd tweaks

### PR DESCRIPTION
**Pull request by anitazha**
- replace errno checks with macros
- added an extended test
- change swap existence check

### CODE REVIEW

File: src/oom/oomd.c [hunk 1]
```diff
@@ -131,8 +133,16 @@ static int run(int argc, char *argv[]) {
...
+
+        r = get_proc_field("/proc/meminfo", "SwapTotal", WHITESPACE, &swap);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get SwapTotal from /proc/meminfo: %m");
+
+        r = safe_atollu(swap, &s);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse SwapTotal from /proc/meminfo: %s: %m", swap);
+        if (s == 0)
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Requires swap to operate");
```
Comment (poettering) [inline src/oom/oomd.c:146]: hmm, on a kernel compiled without swap support, how will the error message look like now? i.e. is SwapTotal in /proc/meminfo even then and hardcoded to 0?

Comment (poettering) [reply src/oom/oomd.c:146]: (if that's the case, maybe add a brief comment clarifying that the field exists even on swap-disabled kernels)

**Comment by anitazha** [comment]
Taking the liberty to update the green label. Added the comment. But if `getconf` isn't available I might have to rework the test more

File: src/oom/test-oomd-util.c [hunk 2]
```diff
@@ -57,7 +57,7 @@ static void test_oomd_cgroup_kill(void) {
 
         /* If we don't have permissions to set xattrs we're likely in a userns or missing capabilities */
         r = cg_set_xattr(SYSTEMD_CGROUP_CONTROLLER, cgroup, "user.oomd_test", "test", 4, 0);
-        if (IN_SET(r, -EPERM, -ENOTSUP))
+        if (ERRNO_IS_PRIVILEGE(r) || ERRNO_IS_NOT_SUPPORTED(r))
                 return (void) log_tests_skipped("Cannot set user xattrs");
```
Comment (yuwata) [inline src/oom/test-oomd-util.c:61]: I found this pattern first time: `return (void) log_xxx(...);`
I prefer `log_xxx(...); return;`

Comment (anitazha) [reply src/oom/test-oomd-util.c:61]: lol you're right why did I do that it's all over the oomd and PSI unit tests... I'll split the lines in another PR so I don't have to restart the CI again

Comment (yuwata) [reply src/oom/test-oomd-util.c:61]: Yeah, let's update it later.

Comment (keszybz) [reply src/oom/test-oomd-util.c:61]: Oh, I didn't even know this is possible. It's a pretty cool trick.

File: test/units/testsuite-56.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Slice=testworkload.slice
+ExecStart=/usr/lib/systemd/tests/testdata/units/testsuite-56-slowgrowth.sh
+EOF
+
+# This generates no memory pressure
+cat > /etc/systemd/system/testchill.service <<EOF
+[Service]
+MemoryHigh=2M
+Slice=testworkload.slice
+ExecStart=sleep infinity
```
Comment (keszybz) [inline test/units/testsuite-56.sh:40]: Those three units are all completely static. I'd prefer to have them as normal files in test/units/. In particular, creating files in /etc/ means that the image the tests are executed from cannot be immutable. /run/systemd/system would be a better place, but static files are even better.

File: test/units/testsuite-56.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+EOF
+
+systemctl daemon-reload
```
Comment (keszybz) [inline test/units/testsuite-56.sh:43]: Once the files are made static, no reload is necessary.

**Comment by keszybz** [comment]
OK, let's merge this. Some fixups can be done later.

**Pull request merged**